### ISSUE TITLE
Rename file by renaming on an aliased path

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#27959]
From https://github.com/Microsoft/vscode/issues/60918

**TypeScript Version:**  3.2.0-dev.20181019

**Search Terms:**
- rename
- path rename

**Code**
For a simple js project

`jsconfig.json`
```json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es2016",
        "jsx": "preserve",
        "checkJs": true,
        "baseUrl": ".",
        "paths": {
            "~/*": [
                "*"
            ]
        }
    },
    "exclude": [
        "node_modules",
        "**/node_modules/*"
    ]
}
```

`main.ts`
```js
import * as foo from '~/src/foo';

console.log(foo);
```

`src/foo.js`
```js
export class Foo { }
```

1. In `main.js`, f2 rename on `foo` in the path

**Expected behavior:**
This should allow renaming `foo.js`

**Actual behavior:**
See error: `you cannot rename a module via a global import`

**Comment by heavenkiller2018** [comment]
is there any plan to fix it?

**Comment by varthc** [comment]
Any update? Thanks!

**Comment by timoisalive** [comment]
Yep, hoping to see a fix to this as well, now that also Expo projects support path aliases (https://docs.expo.dev/guides/typescript/#path-aliases).

**Comment by simplenotezy** [comment]
This have been mentioned many times across different issues, but I don't see any solution. Is there one, or are we just currently doomed if we use aliased imports?

https://github.com/microsoft/vscode/issues/97135

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
